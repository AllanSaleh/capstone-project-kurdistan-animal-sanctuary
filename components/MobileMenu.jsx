import React from "react";
import PropTypes from "prop-types";
import LoginandLanguage from "./LoginandLanguage";
import Navbutton from "./Navbutton";

const MobileMenu = ({ state }) => (
  <div className={state === true ? "block " : "hidden"}>
    <div className="flex felx-row bg-primary">
      <ul className=" px-5 flex flex-col">
        <li className="pb-1">
          <Navbutton name="Home" link="./" className="w-20" />
        </li>

        <li className="pb-1">
          <Navbutton name="Contact" link="./ContactUs" />
        </li>
        <li className="pb-1">
          <Navbutton name="Adopt" link="./AdoptThePerfectPet" />
        </li>
        <li className="pb-1">
          <Navbutton name="Resources" link="./Resource" />
        </li>
      </ul>
      <LoginandLanguage />
    </div>
  </div>
);

MobileMenu.propTypes = {
  state: PropTypes.bool.isRequired,
};

export default MobileMenu;
