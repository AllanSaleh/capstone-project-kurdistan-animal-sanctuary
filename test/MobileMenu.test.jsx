import React from "react";
import renderer from "react-test-renderer";
import { test, expect } from "@jest/globals";
import MobileMenu from "../components/MobileMenu";

test("proparly renders the MobileMenu component", () => {
  const tree = renderer.create(<MobileMenu />).toJSON();
  expect(tree).toMatchSnapshot();
});
