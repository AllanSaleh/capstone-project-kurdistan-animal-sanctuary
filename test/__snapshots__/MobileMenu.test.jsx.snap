// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proparly renders the MobileMenu component 1`] = `
<div
  className="hidden"
>
  <div
    className="flex felx-row bg-primary"
  >
    <ul
      className=" px-5 flex flex-col"
    >
      <li
        className="pb-1"
      >
        <button
          className="mr-3 text-gray-300 hover:text-highlight"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          Home
        </button>
      </li>
      <li
        className="pb-1"
      >
        <button
          className="mr-3 text-gray-300 hover:text-highlight"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          Contact
        </button>
      </li>
      <li
        className="pb-1"
      >
        <button
          className="mr-3 text-gray-300 hover:text-highlight"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          Adopt
        </button>
      </li>
      <li
        className="pb-1"
      >
        <button
          className="mr-3 text-gray-300 hover:text-highlight"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          Resources
        </button>
      </li>
    </ul>
    <span>
      <span
        className="hidden"
      >
        <button
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          <img
            alt="Website logo "
            className="w-16 h-16  cursor-pointer  rounded-full   "
            src="./profile.png"
          />
        </button>
      </span>
      <span
        className="block mt-7 sm:inline-block sm:mt-0 "
      >
        <button
          className="mr-3.5 w-28 h-10 text-highlight border-2 border-highlight rounded-lg "
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          Login
        </button>
        <button
          className=" w-28 h-10 text-primary bg-highlight hover:bg-darken rounded-lg "
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          Sign up
        </button>
      </span>
      <span
        className="bg-secondary text-sm text-white leading-none  rounded-full inline-flex mt-6 mb-4 md:mb-0 sm:mt-0 ml-24  sm:ml-14"
      >
        <button
          className="inline-flex items-center transition-colors duration-300 ease-in focus:outline-none hover:text-black focus:text-primary foucs:bg-primary rounded-l-full px-4 py-2 active"
          type="button"
        >
          <span>
            en
          </span>
        </button>
        <button
          className="inline-flex items-center transition-colors duration-300 ease-in focus:outline-none hover:text-black focus:text-primary foucs:bg-primary rounded-l-full px-4 py-2"
          type="button"
        >
          <span>
            ku
          </span>
        </button>
      </span>
    </span>
  </div>
</div>
`;
